editor/modes/visual-editor/block.js

```diff
@@ -18,13 +19,36 @@ class VisualEditorBlock extends wp.element.Component {
 		this.setAttributes = this.setAttributes.bind( this );
 		this.maybeDeselect = this.maybeDeselect.bind( this );
 		this.maybeHover = this.maybeHover.bind( this );
+		this.onFormatChange = this.onFormatChange.bind( this );
+		this.toggleFormat = this.toggleFormat.bind( this );
 		this.previousOffset = null;
+		this.state = {
+			formats: {}
+		};
 	}
 
 	bindBlockNode( node ) {
 		this.node = node;
 	}
 
+	onFormatChange( formats ) {
+		if ( ! isEqual( formats, this.state.formats ) ) {
```

Comment:
Why would we allow `onFormatChange` to be invoked if a change actually hasn't occurred? Or in any case, why would we optimize for it? Thinking we can drop the `isEqual` check altogether.